spec/controllers/file_sets_controller_spec.rb

```diff
@@ -26,4 +26,23 @@
       expect { get :edit, params: { id: file_set.id.to_s } }.not_to raise_error
     end
   end
+
+  describe "PUT /file_sets/id" do
+    context 'with a derivative service for images in the TIFF' do
+      let(:create_derivatives_class) { class_double(CreateDerivativesJob).as_stubbed_const(transfer_nested_constants: true) }
+      let(:original_file) { instance_double(FileMetadata) }
+      let(:file_set) { FactoryGirl.create_for_repository(:file_set) }
+      before do
+        allow(original_file).to receive(:mime_type).and_return('image/tiff')
+        allow(file_set).to receive(:original_file).and_return(original_file)
+        allow(create_derivatives_class).to receive(:perform_later).and_return(success: true)
+      end
+
+      it "can regenerate derivatives" do
+        put :derivatives, params: { id: file_set.id.to_s }
+
+        expect(response).to redirect_to(file_set)
```

Comment:
Can you add an expectation that the derivatives class has received the perform_later call?